lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,19 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      repo_files = Set.new(Overcommit::GitRepo.all_files)
+      conflict_hash = repo_files.classify(&:downcase).
+        select { |_, files| files.size > 1 }
+      conflict_files = applicable_files.
+        select { |file| conflict_hash.include? file.downcase }
+
+      conflict_files.map do |file|
+        conflicts = conflict_hash[file.downcase].map { |f| File.basename(f) }
+        msg = "Case-insensitivity conflict detected: #{conflicts.join(', ')}"
```

Comment:
I would change this message to emphasize that this is a problem with case-insensitive filesystems.

```
Conflict detected for case-insensitive file systems: file1.txt, File1.txt
```